UNL-CSE-AMBASSADORS/VITA
# [77] Shift Management

This PR creates a new "Site Management" page where a site administrator can create shifts at a site, and create appointment times at a site. In the future, this page will be expanded to include the functionality for editing, adding, and deleting sites. 

## Testing
1. Run `./scripts/build.sh`
2. Run the DB scripts
3. Log in as preparer
    * You should not be able to see the "Site Management" link under the "Management" tab
4. Logout and login as siteadmin
    * You should be able to see the "Site Management" link under the "Management" tab
5. Navigate to the "Site Management" page
    * You should see all 7 sites listed on the left-hand side of the screen
6. Click on "Test Site"
    * Information for "Test Site" should load and be displayed
    * You should see 2 volunteer shifts and 7 appointment times listed
7. Hit the "Add Shift" button
    * A form should show
        * Hitting the cancel button should hide the form
    * Choose a date
        * You should not be able to choose a date in the past
    * Choose a start time
    * Choose an end time
        * The end time should start 30 minutes after the start time
        * If you change the start time, the end time should clear and you will be re-prompted to select a time
    * Hit Save
        * You should get a green success notification
        * The shift should be added to the table of shifts above
8. Refresh the page
    * The shift you just added should be in that table for "Test Site"
9. Hit the "Add Appointment Time" button
    * A form should show
        * Hitting cancel should hide the form
    * You should only be able to select a date that has a shift on it (I suggest choosing the date of the shift you just added above!)
    * You should only be able to choose a time that is within the shifts on that day
        * Note that if there is already an appointment time at a given time, you can't schedule another appointment time then
    * Changing the approximate length in minutes field should update the scheduled times you can choose
    * Hit the Save button
        * You should get a green success notification
        * The appointment time should be added to the table of appointment times above
10. Refresh the page
    * The appointment time you just added should be in that table for "Test Site"
11. Navigate to the profile page
    * Sign up as a preparer for the shift that you just added
        * You should be able to successfully sign up
12. Log out 
13. Navigate to the appointment sign up page
    * NOTE: You will need to remove the comments that are preventing you from being able to sign up for an appointment
    * Sign up for an appointment at the appointment time that you just created (you should be able to do this since you signed up a preparer for this shift time)
        * You should be able to sign up successfully

If all that works, then this piece of functionality should be good!
